src/main/java/org/fogbowcloud/manager/core/datastructures/OrderStateTransitioner.java

```diff
@@ -0,0 +1,73 @@
+package org.fogbowcloud.manager.core.datastructures;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.models.linkedList.SynchronizedDoublyLinkedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+
+public class OrderStateTransitioner {
+
+    private static final Logger LOGGER = Logger.getLogger(OrderStateTransitioner.class);
+
+    /**
+     * Updates the state of the given order. This method is not
+     * thread-safe. The caller must assure that only one
+     * thread can call it at the same time.
+     *
+     *  @param order, the order that will transition through states
+     * @param newState, the new state
+     */
+    public static boolean updateState(Order order, OrderState newState) {
+        OrderState currentState = order.getOrderState();
+
+        SynchronizedDoublyLinkedList origin = getOrdersList(currentState);
+        SynchronizedDoublyLinkedList destination = getOrdersList(newState);
+
+        if (origin == null) {
+            LOGGER.error(String.format("Could not find origin list for state %s", currentState));
+        } else if (destination == null) {
+            LOGGER.error(String.format("Could not find destination list for state %s", newState));
+        } else {
+            if (origin.removeItem(order)) {
+                order.setOrderState(newState);
+                origin.addItem(order);
```

Comment:
It is not origin, but destination. destination.addItem(order);

Reply:
Nice! I hadn't see this mistake.